Sources/NIO/ByteBuffer-conversions.swift

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension Array where Element == UInt8 {
+    
+    /// Creates a `[UInt8]` from the given buffer. The entire readable portion of the buffer will be read.
+    /// - parameter buffer: The buffer to read.
+    init(buffer: ByteBuffer) {
+        self = Array(buffer.readableBytesView)
```

Comment:
I _think_ this generates moderately less efficient code than taking a local `var` and using `readBytes`.